http-client/src/main/java/io/airlift/http/client/jetty/FailedHttpResponseFuture.java

```diff
@@ -0,0 +1,66 @@
+package io.airlift.http.client.jetty;
+
+import com.google.common.util.concurrent.ListenableFuture;
+import io.airlift.http.client.HttpClient;
+
+import java.util.concurrent.ExecutionException;
+import java.util.concurrent.Executor;
+import java.util.concurrent.TimeUnit;
+import java.util.concurrent.TimeoutException;
+
+import static com.google.common.util.concurrent.Futures.immediateFailedFuture;
+
+public class FailedHttpResponseFuture<T>
+        implements HttpClient.HttpResponseFuture<T>
```

Comment:
I think this can be
```java
public class FailedHttpResponseFuture<T>
        extends SimpleForwardingListenableFuture<T>
        implements HttpClient.HttpResponseFuture<T>
{
    public FailedHttpResponseFuture(Throwable throwable)
    {
        super(immediateFailedFuture(throwable));
    }

    @Override
    public String getState()
    {
        return "FAILED";
    }
}

Reply:
NICE